src/core/useSetAtom.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useContext } from 'react'
+import { SECRET_INTERNAL_getScopeContext as getScopeContext } from 'jotai'
+import type { WritableAtom } from 'jotai'
+import type { Atom, Scope, SetAtom } from '../core/atom'
+import { WRITE_ATOM } from '../core/store'
+import type { VersionObject } from '../core/store'
```

Comment:
I believe that things in `core` should not import anything `from 'jotai'`

```suggestion
import { getScopeContext } from './contexts'
import type { Atom, Scope, SetAtom, WritableAtom } from './atom'
import { WRITE_ATOM } from './store'
import type { VersionObject } from './store'
```